shared_model/interfaces/iroha_internal/transaction_sequence_factory.hpp

```diff
@@ -9,17 +9,20 @@
 #include "interfaces/iroha_internal/transaction_sequence.hpp"
 
 #include "common/result.hpp"
-#include "validators/field_validator.hpp"
-#include "validators/transactions_collection/transactions_collection_validator.hpp"
+#include "interfaces/iroha_internal/transaction_batch_factory.hpp"
+#include "validators/default_validator.hpp"
 
 namespace shared_model {
   namespace interface {
 
     /**
      * Provides a method that creates a transaction sequence from  a collection
-     * of transactions. Field validator is used by default
+     * of transactions
      */
     class TransactionSequenceFactory {
+     private:
+      static const std::unique_ptr<TransactionBatchFactory> batch_factory_;
```

Comment:
Looks illegal:
1. Private member in the beginning of the definition
2. Static + unique_ptr
3. Not parametrized in ctor?